code/parachain/frame/composable-traits/src/xcm/assets.rs

```diff
@@ -89,3 +89,38 @@ pub struct ForeignMetadata<AssetNativeLocation> {
 	pub decimals: Option<Exponent>,
 	pub location: AssetNativeLocation,
 }
+
+pub trait PermissionlessAsset {
+	type AccountId;
+	type AssetId;
+	type AssetNativeLocation;
+	type Balance;
+
+	fn register_asset(
+		account_id: Self::AccountId,
+		asset_id: Self::AssetId,
+		location: Self::AssetNativeLocation,
+		name: Option<BiBoundedAssetName>,
+		symbol: Option<BiBoundedAssetSymbol>,
+		decimals: u8,
+		is_cosmwasm: bool,
+	) -> DispatchResult;
+}
+
+pub trait CosmWasmIssuance {
+	type AccountId;
+	type AssetId;
+	type Balance;
+
+	fn mint_into(
+		asset_id: Self::AssetId,
```

Comment:
trait on our side would contain `who` parameters. sure in CW messages it will be not, but taken from contract sender.

Reply:
this is mint by the admin of an asset. i assumed transfer will be handled by assets pallet